controlplane/pkg/apis/connectioncontext/helpers.go

```diff
@@ -30,7 +30,17 @@ func (c *ConnectionContext) IsValid() error {
 	return nil
 }
 
-func (c *ConnectionContext) MeetsRequirements(original *ConnectionContext) error {
+func (c *DNSConfig) Validate() error {
+	if c == nil {
+		return fmt.Errorf("dns config should be nil")
```

Comment:
Do you mean dns config should *not* be nil?

Reply:
Oops! I'll fix in next commit

Reply:
done